notion-core/src/main/scala/com/ambiata/notion/core/LocationIO.scala

```diff
@@ -144,6 +144,32 @@ case class LocationIO(configuration: Configuration, s3Client: AmazonS3Client) {
     case h @ HdfsLocation(path)      => Hdfs.delete(new Path(path)).run(configuration)
   }
 
+  /**
+   * copy a file by simply piping lines from one location to the other
+   *
+   * if overwrite is false and the location file already exists, return an Error
+   */
+  def copyFile(from: Location, to: Location, overwrite: Boolean): RIO[Unit] = {
+    for {
+      exists <- exists(to)
+      _      <-
+      if (exists)
+        if (overwrite) delete(to) >> pipe(from, to)
+        else           RIO.fail(s"There is already a file at ${to.render}")
+      else pipe(from, to)
+    } yield ()
+  }
+
+  /** pipe bytes from one location to another */
+  def pipe(from: Location, to: Location): RIO[Unit] =
```

Comment:
/minor Just the name/comment here doesn't make a lot of sense here any more, it is basically just unsafeCopyFile (unsafe just because it doesn't do the override here. I would consider inlining, but if not, I would rename to something that fits a bit better.